src/components/ViewerBorder/index.jsx

```diff
@@ -1,30 +1,57 @@
 import React from 'react';
+import Fullscreen from 'react-full-screen';
 import PropTypes from 'prop-types';
+import { Button, Icon } from '@edx/paragon';
 
 import './ViewerBorder.scss';
 
+class ViewerBorder extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      isFull: false,
+    };
+  }
 
-const ViewerBorder = props => (
-  <span id={props.spanId}>
-    <div className={`viewer-border${props.spanFullWidth ? ' w-100' : ''}`}>
-      { props.title.trim() &&
-        <div className="grow">
-          <div className="viewer-title" >{props.title} </div>
-        </div>
-      }
-      <div className="viewer-content">
-        {props.children}
-      </div>
-      { props.caption.trim() &&
-        /* eslint-disable react/no-danger */
-        <div className="viewer-caption grow">
-          <div dangerouslySetInnerHTML={{ __html: props.caption }} />
+  goFull = () => {
+    this.setState({ isFull: true });
+  }
+
+  render() {
+    return (
+      <span id={this.props.spanId}>
+        <div className={`viewer-border${this.props.spanFullWidth ? ' w-100' : ''}`}>
+          { this.props.title.trim() &&
+            <div className="grow">
+              <div className="viewer-title" >{this.props.title}
+                <Button
+                  className={['control-btn', 'float-right']}
+                  label={<Icon className="fa fa-expand" />}
+                  onClick={this.goFull}
+                />
+              </div>
+            </div>
+          }
+          <Fullscreen
```

Comment:
You should add a prop to ViewerBorder like
`showFullscreen: PropTypes.bool,` and default it's value to false. Then pass in showFullscreen from JournalPage only for Videos and Documents. We don't want it on images for now.

Reply:
Fullscreen for videos is not working as expected, probably because it's in an iframe. The embedded video component has a fullscreen button on it's bottom toolbar. When I press that button, the video correctly fills the iframe. Then when I press the ViewerBorder fullscreen button the entire window goes fullscreen but the bottom part of the video is cutoff. Can't see the bottom toolbar.

Reply:
@bill-filler for some reason github was not showing my last commit has to close and re-open this PR.

Reply:
@bill-filler for some reason github is not showing my last commit.